### ISSUE TITLE
Multiple `has_uploadcare_file` calls overwrite the previous ones

### ISSUE THREAD

**Issue by Oshuma** [uploadcare/uploadcare-rails#61]
If you have more than one `has_uploadcare_file` calls in a single model (even with different attributes), only the last one loaded works.  All other attributes will return the file data for the last attribute in the 'list'.  Example:

```ruby
class Album < ActiveRecord::Base
  has_uploadcare_file :album_art
  has_uploadcare_file :album_art_back
end
```

```
[1] pry(main)> album = Album.first
=> #<Album:0x0055fb6d738728
 id: 1,
 album_art: "https://ucarecdn.com/22366021-02c5-42d6-a598-8e85e9c18d16/",
 album_art_back: "https://ucarecdn.com/8c38cb9b-9c1e-41ad-b8b7-b39cfebbb9af/">
[2] pry(main)> album.album_art
Cache read: https://ucarecdn.com/8c38cb9b-9c1e-41ad-b8b7-b39cfebbb9af/
D, [2018-01-26T15:11:00.354794 #15215] DEBUG -- : Cache read: https://ucarecdn.com/8c38cb9b-9c1e-41ad-b8b7-b39cfebbb9af/
Dalli::Server#connect 127.0.0.1:11211
D, [2018-01-26T15:11:00.354995 #15215] DEBUG -- : Dalli::Server#connect 127.0.0.1:11211
=> #<Uploadcare::Rails::File uuid="8c38cb9b-9c1e-41ad-b8b7-b39cfebbb9af", operations=[], datetime_removed=nil, datetime_stored="2018-01-26T19:08:53.415735Z", datetime_uploaded="2018-01-26T19:08:53.109187Z", image_info={"orientation"=>nil, "format"=>"JPEG", "height"=>500, "width"=>500, "geo_location"=>nil, "datetime_original"=>nil, "dpi"=>nil}, is_image=true, is_ready=true, mime_type="image/jpeg", original_file_url="https://ucarecdn.com/8c38cb9b-9c1e-41ad-b8b7-b39cfebbb9af/mbidcb100c18fb2948288d242fdbac6d6cff15675237690_thumb500.jpg", original_filename="mbid-cb100c18-fb29-4828-8d24-2fdbac6d6cff-15675237690_thumb500.jpg", size=70552, url="https://api.uploadcare.com/files/8c38cb9b-9c1e-41ad-b8b7-b39cfebbb9af/", source=nil>
[3] pry(main)> album.album_art.url
Cache read: https://ucarecdn.com/8c38cb9b-9c1e-41ad-b8b7-b39cfebbb9af/
D, [2018-01-26T15:11:02.806458 #15215] DEBUG -- : Cache read: https://ucarecdn.com/8c38cb9b-9c1e-41ad-b8b7-b39cfebbb9af/
=> "https://ucarecdn.com/8c38cb9b-9c1e-41ad-b8b7-b39cfebbb9af/"
[4] pry(main)> album.album_art_back
Cache read: https://ucarecdn.com/8c38cb9b-9c1e-41ad-b8b7-b39cfebbb9af/
D, [2018-01-26T15:11:08.642813 #15215] DEBUG -- : Cache read: https://ucarecdn.com/8c38cb9b-9c1e-41ad-b8b7-b39cfebbb9af/
=> #<Uploadcare::Rails::File uuid="8c38cb9b-9c1e-41ad-b8b7-b39cfebbb9af", operations=[], datetime_removed=nil, datetime_stored="2018-01-26T19:08:53.415735Z", datetime_uploaded="2018-01-26T19:08:53.109187Z", image_info={"orientation"=>nil, "format"=>"JPEG", "height"=>500, "width"=>500, "geo_location"=>nil, "datetime_original"=>nil, "dpi"=>nil}, is_image=true, is_ready=true, mime_type="image/jpeg", original_file_url="https://ucarecdn.com/8c38cb9b-9c1e-41ad-b8b7-b39cfebbb9af/mbidcb100c18fb2948288d242fdbac6d6cff15675237690_thumb500.jpg", original_filename="mbid-cb100c18-fb29-4828-8d24-2fdbac6d6cff-15675237690_thumb500.jpg", size=70552, url="https://api.uploadcare.com/files/8c38cb9b-9c1e-41ad-b8b7-b39cfebbb9af/", source=nil>
[5] pry(main)> album.album_art_back.url
Cache read: https://ucarecdn.com/8c38cb9b-9c1e-41ad-b8b7-b39cfebbb9af/
D, [2018-01-26T15:11:10.793781 #15215] DEBUG -- : Cache read: https://ucarecdn.com/8c38cb9b-9c1e-41ad-b8b7-b39cfebbb9af/
=> "https://ucarecdn.com/8c38cb9b-9c1e-41ad-b8b7-b39cfebbb9af/"
```

As you can see, the `#album_art_back` attribute returns info from the `#album_art` attribute, even though the attribute value is stored correctly in the database.  It looks like the problem is with the dynamic `#build_file` method in `Uploadcare::Rails::ActiveRecord#has_uploadcare_file` (and probably the same issue with that dynamic method in `#has_uploadcare_group`).

I've changed that method definition to use the attribute like so:
```ruby
define_method "build_#{attribute}_file"
```
which seems to fix the issue.  I'll submit a PR for this shortly.

**Comment by Oshuma** [comment]
Can I get an update on this PR?  We've having to use my fork which includes the fix in our app, since we have a model that has two UC fields.

**Comment by vizvamitra** [comment]
@Oshuma Hello and thank you for your PR! Last couple of months I'm having troubles with spare time to work on this gem, but I'll find some and review your PR in a day or two

**Comment by rastyagaev** [comment]
Seems like fixed in #74
